manage_people/views.py

```diff
@@ -400,26 +423,28 @@ def add_member_to_course(user_id, user_role_id, course_instance_id,
         #       in the future.
         canvas_role_name = get_canvas_role_name(user_role_id)
         canvas_section = get_canvas_course_section(course_instance_id)
-        if canvas_section:
-            canvas_enrollment = add_canvas_section_enrollee(
-                canvas_section['id'], canvas_role_name, user_id, enrollment_role_id=user_role.canvas_role_id)
-        else:
-            canvas_enrollment = add_canvas_course_enrollee(
-                canvas_course_id, canvas_role_name, user_id, enrollment_role_id=user_role.canvas_role_id)
-
-        if canvas_enrollment:
-            # flush the canvas api caches on successful enrollment
-            canvas_api_helper_courses.delete_cache(
-                canvas_course_id=canvas_course_id)
-            canvas_api_helper_enrollments.delete_cache(canvas_course_id)
-            canvas_api_helper_sections.delete_cache(canvas_course_id)
-        else:
-            logger.error(
-                f'Unable to enroll {user_id} as user_role_id {user_role_id} '
-                f'(Canvas role id {user_role.canvas_role_id}) for course instance id {course_instance_id}.'
-            )
-    return existing_enrollment, person
+        try: 
+            if canvas_section:
+                canvas_enrollment = add_canvas_section_enrollee(
+                    canvas_section['id'], canvas_role_name, user_id, enrollment_role_id=user_role.canvas_role_id)
+            else:
+                canvas_enrollment = add_canvas_course_enrollee(
+                    canvas_course_id, canvas_role_name, user_id, enrollment_role_id=user_role.canvas_role_id)
+
+            if canvas_enrollment:
+                # flush the canvas api caches on successful enrollment
+                canvas_api_helper_courses.delete_cache(
+                    canvas_course_id=canvas_course_id)
+                canvas_api_helper_enrollments.delete_cache(canvas_course_id)
+                canvas_api_helper_sections.delete_cache(canvas_course_id)
+            else:
+                error_message = f"Failed to enroll user {user_id} in Canvas with role ID {user_role.canvas_role_id}."
+                logger.warning(error_message)
+        except Exception as e:
+            error_message = f"Canvas enrollment failed for user {user_id}: {str(e)}"
+            logger.exception(error_message)
```

Comment:
Since this is an unexpected error, I think it'd be helpful to have more info for troubleshooting. I think Chris suggested something similar above, but I'd consider adding more request data to the log statement via the `extra` kwarg (e.g. including `course_instance_id` and `canvas_course_id`, as those should be available as arguments within this function).